notebook/terminal/__init__.py

```diff
@@ -7,36 +7,38 @@
     raise ImportError("terminado >= 0.8.1 required, found %s" % terminado.__version__)
 
 from ipython_genutils.py3compat import which
-from terminado import NamedTermManager
 from tornado.log import app_log
 from notebook.utils import url_path_join as ujoin
+from .terminalmanager import TerminalManager
 from .handlers import TerminalHandler, TermSocket
 from . import api_handlers
 
-def initialize(webapp, notebook_dir, connection_url, settings):
+
+def initialize(parent):
```

Comment:
This is nitpicky, but I think this argument should be `nbapp` or something similar. It is expected to be *specifically* a NotebookApp instance—all the attributes accessed below are unique to notebookapps.